packages/@aws-cdk/aws-gamelift/README.md

```diff
@@ -110,6 +103,261 @@ new gamelift.Script(this, 'Script', {
+      parameters: '-logFile /local/game/logs/myserver1935.log -port 1935',
+      concurrentExecutions: 100,
+    }]
+  }
+});
+```
+
+See [Managing how game servers are launched for hosting](https://docs.aws.amazon.com/gamelift/latest/developerguide/fleets-multiprocess.html)
+in the *Amazon GameLift Developer Guide*.
+
+### Defining an instance type
+
+GameLift uses Amazon Elastic Compute Cloud (Amazon EC2) resources, called
+instances, to deploy your game servers and host game sessions for your players.
+When setting up a new fleet, you decide what type of instances your game needs
+and how to run game server processes on them (using a runtime configuration). All instances in a fleet use the same type of resources and the same runtime
+configuration. You can edit a fleet's runtime configuration and other fleet
+properties, but the type of resources cannot be changed.
+
+```ts
+declare const build: gamelift.Build;
+new gamelift.BuildFleet(this, 'Game server fleet', {
+  fleetName: 'test-fleet',
+  content: build,
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.C5, ec2.InstanceSize.LARGE),
+  runtimeConfiguration: {
+    serverProcesses: [{
+      launchPath: '/local/game/GameLiftExampleServer.x86_64',
+    }]
+  }
+});
+```
+
+### Using Spot instances
+
+When setting up your hosting resources, you have the option of using Spot
+Instances, On-Demand Instances, or a combination.
+
+By default, fleet are using on demand capacity.
+
+```ts
+declare const build: gamelift.Build;
+new gamelift.BuildFleet(this, 'Game server fleet', {
+  fleetName: 'test-fleet',
+  content: build,
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.C4, ec2.InstanceSize.LARGE),
+  runtimeConfiguration: {
+    serverProcesses: [{
+      launchPath: '/local/game/GameLiftExampleServer.x86_64',
+    }]
+  },
+  useSpot: true
+});
+```
+
+### Allowing Ingress traffic
+
+The allowed IP address ranges and port settings that allow inbound traffic to
+access game sessions on this fleet.
+
+New game sessions are assigned an IP address/port number combination, which
+must fall into the fleet's allowed ranges. Fleets with custom game builds must
+have permissions explicitly set. For Realtime Servers fleets, GameLift
+automatically opens two port ranges, one for TCP messaging and one for UDP.
+
+```ts
+declare const build: gamelift.Build;
+
+const fleet = new gamelift.BuildFleet(this, 'Game server fleet', {
+  fleetName: 'test-fleet',
+  content: build,
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.C4, ec2.InstanceSize.LARGE),
+  runtimeConfiguration: {
+    serverProcesses: [{
+      launchPath: '/local/game/GameLiftExampleServer.x86_64',
+    }]
+  },
+  ingressRules: [{
+    source: gamelift.Peer.anyIpv4(),
+    port: gamelift.Port.tcpRange(100, 200),
+  }]
+});
+// Allowing all IP Addresses from port 1111 to port 1122 on TCP Protocol
+fleet.addIngressRule(gamelift.Peer.anyIpv4(), gamelift.Port.tcpRange(100, 200));
```

Comment:
I guess this example is redundant, since this ingress rule was already passed to the constructor. So maybe you should remove this to avoid confusion. In any case, what happens if you define the same rule twice, as in this example? is it idempotent?

Reply:
Ok I will remove that line.

If the same rule is added twice, it will create two different ingress rule on the fleet.